packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/representations/RepresentationService.java

```diff
@@ -166,12 +165,13 @@ public void delete(UUID representationId) {
 
     @Override
     public boolean isDangling(IEditingContext editingContext, IRepresentation representation) {
-        if (representation instanceof ISemanticRepresentation semanticRepresentation) {
-            String targetObjectId = semanticRepresentation.getTargetObjectId();
-            Optional<Object> optionalObject = this.objectSearchService.getObject(editingContext, targetObjectId);
-            return optionalObject.isEmpty();
+        String targetObjectId = representation.getTargetObjectId();
+        if (Objects.equals(targetObjectId, editingContext.getId())) {
+            // Transient representation are never candidate for explicit removal
```

Comment:
I'll update that since the details view, the representations view or the related elements view are all transient but they can be candidate for removal.

Reply:
This also makes me realize that I've never handled dangling representations in the new runtime, I'll fix that too.